### PR TITLE
[BEAM-823, BEAM-923] Add localhost option for DatastoreIO

### PR DESCRIPTION
**Pull request by vikkyrk**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
- Verified with Datastore emulator locally

### CODE REVIEW

**Comment by vikkyrk** [comment]
R: @dhalperi

**Comment by vikkyrk** [comment]
R: @davorbonaci since Dan is busy.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java [hunk 1]
```diff
@@ -124,8 +125,10 @@
  *
  * <p><b>Note:</b> Normally, a Cloud Dataflow job will read from Cloud Datastore in parallel across
  * many workers. However, when the {@link Query} is configured with a limit using
- * {@link com.google.datastore.v1.Query.Builder#setLimit(Int32Value)}, then
- * all returned results will be read by a single Dataflow worker in order to ensure correct data.
+ * {@link com.google.datastore.v1.Query.Builder#setLimit(Int32Value)} or if the Query contains
+ * inequality filters like {@code GREATER_THAN, LESS_THAN} etc., then all returned results
+ * will be read by a single Dataflow worker in order to ensure correct data. Since data is read from
```
Comment (davorbonaci) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java]: Please abstract Dataflow away.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java [hunk 2]
```diff
@@ -243,6 +251,7 @@
       abstract Builder setQuery(Query query);
       abstract Builder setNamespace(String namespace);
       abstract Builder setNumQuerySplits(int numQuerySplits);
+      abstract Builder setLocalhost(String hostPort);
```
Comment (davorbonaci) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java]: why is hostPort a String?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java [hunk 3]
```diff
@@ -410,10 +419,16 @@ static RunQueryRequest makeRequest(Query query, @Nullable String namespace) {
           .build();
     }
 
+    public DatastoreV1.Read withLocalhost(String localhost) {
```
Comment (davorbonaci) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java]: I'm not sure what localhost parameter here means.

**Comment by dhalperi** [comment]
Thanks @davorbonaci ;

R: -@dhalperi

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java (hunk 1 above)
Comment (vikkyrk) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java]: Fixed

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java (hunk 2 above)
Comment (vikkyrk) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java]: I would have used 'InetAddress' but to keep it in sync with the DatastoreOptions which uses a String, I kept it consistent.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java (hunk 3 above)
Comment (vikkyrk) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java]: Added comments.

**Comment by vikkyrk** [comment]
R: @davorbonaci  addressed comments.

**Comment by vikkyrk** [comment]
This needs to be rebased after template is merged https://github.com/apache/incubator-beam/pull/1369

**Comment by dhalperi** [comment]
Status?

**Comment by vikkyrk** [comment]
This might have merge conflict with https://github.com/apache/incubator-beam/pull/1369, so planning to serialize them.

**Comment by dhalperi** [comment]
Kicked off postcommit: https://builds.apache.org/view/Beam/job/beam_PostCommit_Java_MavenInstall/2522/org.apache.beam$beam-sdks-java-io-google-cloud-platform/testReport/

and it passed (the GCP-IO ITs ran and passed, at least). So merging.

Thanks!

**Pull request closed without merging**